src/core/dbus-util.h

```diff
@@ -314,7 +314,7 @@
                         return r;                                       \
                                                                         \
                 v = scale(raw, UINT32_MAX);                             \
-                if (v <= 0 || v >= UINT64_MAX)                          \
+                if (!check(v) || v >= UINT64_MAX)                       \
```

Comment:
This relies on the fact that `check(v)` returns `v` if `check` is not defined. So it is equivalent to `v > 0`, or `!v`, and the result is correct. But I think this is just too much magic. I'd just prefer to have the lower limit just passed in as a parameter to `BUS_DEFINE_SET_CGROUP_SCALE`, or maybe to always give the `check` parameter.

The code was already doing that before in BUS_DEFINE_SET_CGROUP_WEIGHT, but I think the code should be reworked to avoid this pattern.

I think it'd also make a lot of sense to move BUS_DEFINE_SET_CGROUP_* macros to the .c file. No need to have them in the header.